chain-vote/src/gang/p256k1.rs

```diff
@@ -57,12 +57,12 @@ impl GroupElement {
         }
     }
 
-    pub fn sum<'a, I>(mut i: I) -> Self
+    pub fn sum<'a, I>(i: I) -> Self
     where
         I: Iterator<Item = &'a Self>,
     {
         let mut sum = GroupElement::zero();
-        while let Some(v) = i.next() {
+        for v in i {
             sum = &sum + v;
```

Comment:
Another sideline note: the `Add` trait impl might take the LHS by value to be less allocatey and more convenient to use. See it's implemented in `String`.